R/CibersortX.R

```diff
@@ -322,12 +351,24 @@ transform_and_save_bulk <- function(bulk, path, verbose = FALSE) {
 #' @return A valid docker command to be run.
 #'
 create_docker_command <- function(in_dir, out_dir,
+                                  container = c("docker", "singularity"),
                                   method = c("create_sig", "impute_cell_fractions"),
                                   verbose = FALSE, ...) {
-  base <- paste0(
-    "docker run -v ", in_dir, ":/src/data:z -v ", out_dir,
-    ":/src/outdir:z cibersortx/fractions --single_cell TRUE"
-  )
+
+  if(container=='docker'){
+    base <- paste0(
+      "docker run -v ", in_dir, ":/src/data:z -v ", out_dir,
+      ":/src/outdir:z cibersortx/fractions --single_cell TRUE"
+    )
+  } else {
+    base <- paste0(
+      "singularity exec -c -B ",in_dir,
```

Comment:
i'd recommend adding a `--no-home` flag. Singularity automatically mounts the home directory and this can sometimes make trouble.

Reply:
solved in [2ced92d](https://github.com/omnideconv/omnideconv/pull/115/commits/2ced92d6d79850ddd3508dbe9f18da87ec162a08)